### PR TITLE
Fix discovery to not update existing konflux configs

### PR DESCRIPTION
**Pull request by creydr**
This should help on https://github.com/openshift-knative/hack/pull/368#discussion_r1814799608

### CODE REVIEW

File: pkg/discover/discover.go [hunk 1]
```diff
@@ -113,6 +113,11 @@ func discover(ctx context.Context, path string) error {
 								Enabled: true,
 							}
 						} else {
+							// copy the Konflux attribute as this is a pointer, and
+							// it would otherwise update the existing branch config
+							v := *other.Konflux
+							other.Konflux = &v
```
Comment (mgencur) [inline pkg/discover/discover.go:119]: Perhaps we should do the same with `other.Prowgen` here? and also on line https://github.com/openshift-knative/hack/pull/384/files#diff-493c512c6574cbf0c3e3e1a06288a2f10b8f8124cb27b95a29cf8ef4e48b6f0fR139 ?

Comment (pierDipi) [reply pkg/discover/discover.go:119]: or maybe worth looking at some deep copy libraries?

Comment (creydr) [reply pkg/discover/discover.go]: :+1: 
updated

**Comment by creydr** [comment]
@mgencur @pierDipi could you recheck?

File: pkg/discover/discover.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import (
 	"encoding/json"
 	"flag"
 	"fmt"
+	"github.com/jinzhu/copier"
```
Comment (pierDipi) [inline pkg/discover/discover.go:8]: sort import

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**